homeassistant/components/media_player/firetv.py

```diff
@@ -66,82 +71,82 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     host = '{0}:{1}'.format(config[CONF_HOST], config[CONF_PORT])
 
-    if CONF_ADBKEY in config:
-        ftv = FireTV(host, config[CONF_ADBKEY])
-        adb_log = " using adbkey='{0}'".format(config[CONF_ADBKEY])
+    if CONF_ADB_SERVER_IP not in config:
+        # "python-adb"
+        if CONF_ADBKEY in config:
+            ftv = FireTV(host, config[CONF_ADBKEY])
+            adb_log = " using adbkey='{0}'".format(config[CONF_ADBKEY])
+        else:
+            ftv = FireTV(host)
+            adb_log = ""
     else:
-        ftv = FireTV(host)
-        adb_log = ""
+        # "pure-python-adb"
+        ftv = FireTV(host, adb_server_ip=config[CONF_ADB_SERVER_IP],
+                     adb_server_port=config[CONF_ADB_SERVER_PORT])
+        adb_log = " using ADB server at {0}:{1}".format(
+            config[CONF_ADB_SERVER_IP], config[CONF_ADB_SERVER_PORT])
 
     if not ftv.available:
         _LOGGER.warning("Could not connect to Fire TV at %s%s", host, adb_log)
         return
 
     name = config[CONF_NAME]
-    get_source = config[CONF_GET_SOURCE]
     get_sources = config[CONF_GET_SOURCES]
 
-    device = FireTVDevice(ftv, name, get_source, get_sources)
+    device = FireTVDevice(ftv, name, get_sources)
     add_entities([device])
     _LOGGER.debug("Setup Fire TV at %s%s", host, adb_log)
 
 
 def adb_decorator(override_available=False):
-    """Send an ADB command if the device is available and not locked."""
-    def adb_wrapper(func):
+    """Send an ADB command if the device is available and catch exceptions."""
+    def _adb_decorator(func):
         """Wait if previous ADB commands haven't finished."""
         @functools.wraps(func)
-        def _adb_wrapper(self, *args, **kwargs):
+        def __adb_decorator(self, *args, **kwargs):
```

Comment:
Don't start with double underscore.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/21221/commits/e66f2403b9756b990a108d6c39ef137e05e6a018